lib/fs.js

```diff
@@ -1921,79 +1921,6 @@ WriteStream.prototype.close = ReadStream.prototype.close;
 // There is no shutdown() for files.
 WriteStream.prototype.destroySoon = WriteStream.prototype.end;
 
-
-// SyncWriteStream is internal. DO NOT USE.
-// Temporary hack for process.stdout and process.stderr when piped to files.
-function SyncWriteStream(fd, options) {
-  Stream.call(this);
-
-  options = options || {};
-
-  this.fd = fd;
-  this.writable = true;
-  this.readable = false;
-  this.autoClose = options.autoClose === undefined ? true : options.autoClose;
-}
-
-util.inherits(SyncWriteStream, Stream);
-
-
-// Export
-Object.defineProperty(fs, 'SyncWriteStream', {
-  configurable: true,
-  writable: true,
-  value: SyncWriteStream
-});
```

Comment:
The "don't use this" warning, the clear indication of it being a hack, the
fact that it was never documented, the fact that it was specifically made
non-enumerable , and the fact that it could be easily duplicated in
userland all point to something that could be easily and justifiably made
internal only and eventually dropped if possible.
On May 14, 2016 2:44 PM, "Jeremiah Senkpiel" <notifications@github.com>
wrote:

> In lib/fs.js
> <https://github.com/nodejs/node/pull/6749#discussion_r63282524>:
>
> > -
> > -  this.fd = fd;
> > -  this.writable = true;
> > -  this.readable = false;
> > -  this.autoClose = options.autoClose === undefined ? true : options.autoClose;
> > -}
> > -
> > -util.inherits(SyncWriteStream, Stream);
> > -
> > -
> > -// Export
> > -Object.defineProperty(fs, 'SyncWriteStream', {
> > -  configurable: true,
> > -  writable: true,
> > -  value: SyncWriteStream
> > -});
>
> I have to wonder if it isn't actually useful in those cases? Why would
> someone ever use this if something didn't specifically need it? idk..
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/6749/files/1173d733a455187d47f4e4eb61a815138e2abfe0#r63282524>
>